R/error_checking.R

```diff
@@ -0,0 +1,340 @@
+#' @param map_legend_title (Character) - Text for the legend header
+#' @param map_x_label (Character) - Label for x axis (default Lon)
+#' @param map_y_label (Character) - Label for y axis (default Lat)
+verify_map_params <- function(bin_method, bins, dpi, expand_xy, map_xy_min_max , map_title, map_palette, map_palette_reverse,
+                               map_palette_type, map_width_height_in, map_legend_title, map_x_label, map_y_label)
+{
+    output <- "Success"
+
+    # Verify bin_method
+    if(is.null(bin_method) || !"character" %in% class(bin_method) || !bin_method %in% c("quantile", "pretty", "equal"))
+    {
+        return("Error: bin_method argument must be one of 'quantile', 'pretty', 'equal'")
+    }
+
+    # Verify bins
+    if(is.null(bins) || class(bins) != "numeric")
+    {
+        return("Error: bins argument must be a positive integer and not NULL")
+    }
+    if(bins < 1 || bins > 100)
+    {
+        return("Error: bins argument must be a valid integer between 1-100")
+    }
+
+    # verify dpi
+    if(!"numeric" %in% (class(dpi)) || dpi < 30 || dpi > 300)
+    {
+        return("Error: DPI argument must be a numeric value between 30 and 300")
+    }
+
+    # Verify expand_xy
+    if(!"numeric" %in% (class(expand_xy)) || length(expand_xy) != 2)
+    {
+        return("Error: expand_xy argument must be a numeric vector of length 2")
+    }
+
+    # Verify map_xy_min_max
+    if(!"numeric" %in% (class(map_xy_min_max)) || length(map_xy_min_max) != 4)
+    {
+        return("Error: map_xy_min_max argument must be a numeric vector of length 4")
+    }
+
+    # Verify title
+    if(!"character" %in% class(map_title) && !is.null(map_title))
+    {
+        return("Error: map_title must be of class character")
+    }
+
+    # Verify map_palette
+    if(!is.null(map_palette) && (!"character" %in% class(map_palette) || !map_palette %in% row.names(brewer.pal.info)) )
+    {
+        return("Error: map_palette must be of class character and a valid entry in the RColorBrewer palette (see brewer.pal.info")
+    }
+
+    # Verify palette_reverse
+    if(!"logical" %in% class(map_palette_reverse))
+    {
+        return("Error: map_palette_reverse must be of type logical (TRUE or FALSE)")
+    }
+
+    # Verify map_width_height_in
+    if(!"numeric" %in% (class(map_width_height_in)) || length(map_width_height_in) != 2)
+    {
+        return("Error: map_width_height_in argument must be a numeric vector of length 2")
+    }
+
+    # Verify map_legend_title
+    if(!"character" %in% class(map_legend_title) && !is.null(map_legend_title))
+    {
+        return("Error: map_legend_title must be of class character or NULL")
+    }
+
+    # Verify map_x_label
+    if(!"character" %in% class(map_x_label)  && !is.null(map_x_label))
+    {
+        return("Error: map_x_label must be of class character or NULL")
+    }
+
+    # Verify map_y_label
+    if(!"character" %in% class(map_y_label) && !is.null(map_y_label))
```

Comment:
Think about setting `NULL` as the default for most of your arguments

Reply:
Went ahead and made any non null arguments to have a default NULL value in this commit cbaedae3b70be1265b8f2e6b1c93ed0f71db95fc